modules/sdk/src/main/scala/org/tessellation/sdk/domain/cluster/programs/Joining.scala

```diff
@@ -153,6 +161,15 @@ sealed abstract class Joining[F[_]: Async: GenUUID: SecurityProvider: KryoSerial
       _ <- verifySignRequest(signRequest, signedSignRequest, PeerId._Id.get(withPeer.id))
         .ifM(Applicative[F].unit, HandshakeSignatureNotValid.raiseError[F, Unit])
 
+      _ <- if (skipJoinRequest) {
+        Applicative[F].unit
+      } else {
+        cluster.getRegistrationRequest
+          .map(JoinRequest.apply)
+          .flatMap(signClient.joinRequest(_).run(withPeer))
+          .ifM(Applicative[F].unit, new Throwable("Unexpected error occured.").raiseError[F, Unit])
```

Comment:
You can use more descriptive error, like "Unexpected error occurred when joining with peer ${withPeer.id}"